### PR TITLE
Docker

### PR DESCRIPTION
**Pull request by Pseudonick47**

### CODE REVIEW

File: ormconfig.json [hunk 1]
```diff
@@ -1,8 +1,11 @@
...
+    "port": 3306,
+    "username": "nest",
+    "password": "nest",
+    "database": "nest",
+    "entities": ["src/modules/**/*.entity{.ts,.js}"],
     "synchronize": true
-}
\ No newline at end of file
+}
+
```
Comment (Pseudonick47) [inline ormconfig.json]: this file is probably redundant

File: README.md [hunk 2]
```diff
@@ -18,37 +18,37 @@
...
 # e2e tests
-$ npm run test:e2e
+$ yarn test:e2e
 
 # test coverage
-$ npm run test:cov
+$ yarn test:cov
 ```
 
 ## Environment Configuration
```
Comment (Pseudonick47) [inline README.md]: TODO: add documentation for docker startup

File: docker-compose-swarm.yml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    deploy:
+      replicas: 1
+      restart_policy:
+        condition: on-failure
+    volumes:
+      - .env:/app/.env
+    depends_on:
+      - db
+
+  nginx:
```
Comment (Pseudonick47) [inline docker-compose-swarm.yml]: TODO: add volume for file uploads, and bind it to all services where needed in all docker-composes

File: docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+services:
+  nest:
+    build: .
+    container_name: nest
+    depends_on:
+      - db
+    volumes:
+      - ./src:/app/src
+      - .env:/app/.env
+      - ./ormconfig.json:/app/ormconfig.json
```
Comment (Pseudonick47) [inline docker-compose.yml]: this is redundant?

File: docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+  nginx:
+    build:
+      context: .
+      dockerfile: Dockerfile-nginx
+    container_name: nest-nginx
```
Comment (Pseudonick47) [inline docker-compose.yml]: unify container names 
either nest nginx, or nest-api, nest-nginx

File: docker/vhost.template [hunk 6]
```diff
@@ -0,0 +1,19 @@
+# vim: ft=nginx
+
+server {
+	listen 80;
+	server_name ${NGINX_SERVER_NAME};
+	root /app/public;
+	client_max_body_size ${NGINX_MAX_BODY};
+
+	location / {
+		# try_files $uri =404;
```
Comment (Pseudonick47) [inline docker/vhost.template]: TODO: add `/storage` location to serve static files

File: Dockerfile-prod [hunk 7]
```diff
@@ -0,0 +1,14 @@
+FROM node:lts-alpine as builder
```
Comment (Pseudonick47) [inline Dockerfile-prod]: remove naming

File: Dockerfile-prod [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+
+WORKDIR /app
+
+ENV NODE_ENV production
+COPY package.json yarn.lock ./
+RUN set -x && yarn
+
+COPY . /app
```
Comment (Pseudonick47) [inline Dockerfile-prod]: COPY . .

File: README.md (hunk 2 above)
Comment (goranjviv) [reply README.md]: added, check if good enough

**Pull request merged**